source/fundamentals/enterprise-authentication.txt

```diff
@@ -323,6 +323,54 @@ see the corresponding syntax.
             .WithMechanismProperty("TOKEN_RESOURCE", "<audience>"); 
          var client = new MongoClient(mongoClientSettings);
 
+.. _csharp-mongodb-oidc-gcp-imds:
+
+GCP IMDS
+~~~~~~~~
+
+If your application runs on a GCP VM, or otherwise uses the
+`GCP Instance Metadata Service <https://cloud.google.com/compute/docs/metadata/querying-metadata>`__,
+you can authenticate to MongoDB by using the {+driver-short+}'s built-in GCP
+support.
+
+You can specify GCP IMDS OIDC authentication on a ``MongoClientSettings`` object either by 
+using a ``MongoCredential`` object or as part of the connection string. Select the 
+:guilabel:`Connection String` or :guilabel:`MongoCredential` tab to
+see the corresponding syntax.
+
+.. tabs::
+
+   .. tab:: Connection String
+      :tabid: mongodb-gcp-imds-connection-string
+    
+      The following code example shows how to specify GCP IMDS OIDC authentication. 
+      Replace the ``<percent-encoded audience>`` placeholder with the percent-encoded
```

Comment:
Question: Does the audience have to be percent-encoded? The [PyMongo driver](https://jira.mongodb.org/browse/DOCSP-40004) currently does not require percent-encoding for the audience value so it would be great to clarify whether this driver also does not require percent-encoding.

Related to percent-encoding, if the driver does not require percent-encoding, specify whether authMechanismProperties values with commas must be passed in through the MongoCredential object rather than the connection string, also based on the [PyMongo driver](https://jira.mongodb.org/browse/DOCSP-40004) ticket.

Reply:
C: Looks like this is [in progress](https://jira.mongodb.org/browse/CSHARP-5106) so you may not need to change it now but keep it in mind.

Reply:
Thanks for bringing this up. Confirmed that the C# driver supports both ways. I'm going to remove "percent-encoded" language to simplify things for users.